packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -146,17 +148,18 @@ export function createPresenceComponent<MotionParams extends Record<string, Moti
           const atoms = visible ? presenceMotion.enter : presenceMotion.exit;
 
           const direction: PresenceDirection = visible ? 'enter' : 'exit';
-          const forceFinishMotion = !visible && isFirstMount;
+          const skipAnimationOnFirstMount = !visible && isFirstMount;
+          const forceFinishMotion = disableMotions || skipAnimationOnFirstMount;
```

Comment:
```suggestion
          const skipAnimationOnFirstMount = !visible && isFirstMount;
          const skipAnimation = disableMotions || skipAnimationOnFirstMount;
```

nit let's be consistent in naming